addons/stock/models/stock_orderpoint.py

```diff
@@ -156,8 +156,11 @@ def _compute_lead_days(self):
             if not orderpoint.product_id or not orderpoint.location_id:
                 orderpoint.lead_days_date = False
                 continue
-            lead_days, dummy = orderpoint.rule_ids._get_lead_days(orderpoint.product_id)
-            lead_days_date = fields.Date.today() + relativedelta.relativedelta(days=lead_days)
+            rules = orderpoint.rule_ids
+            buy_rule = rules.filtered(lambda r: r.action == "buy")
+            seller = orderpoint.product_id.with_company(buy_rule.company_id)._select_seller(quantity=None)
```

Comment:
here we can be a few rules having a buy rule
then can they belong to different companies or company-dependent functionality will cover this?
if not you can have recordset of rules belonging to different companies, if yes then you don't need to filter for buy rule we can use company from the first one rule